aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -23,4 +24,128 @@ Tensor& bmm_out_cuda(Tensor &result, const Tensor& batch1, const Tensor& batch2)
   return legacy::cuda::_th_bmm_out(result, batch1, batch2);
 }
 
+Tensor prepare_matrix_for_cublas(Tensor& tensor, bool& transpose_tensor) {
+  Tensor tensor_;
+  IntArrayRef tensor_strides = tensor.strides();
+
+  if ((tensor_strides[0] == 1) && (tensor_strides[1] != 0)) {
+    tensor_ = tensor;
+    transpose_tensor = false;
+  } else if ((tensor_strides[1] == 1) && (tensor_strides[0] != 0)) {
+    tensor_ = tensor;
+    transpose_tensor = true;
+  } else {
+    transpose_tensor = true;
```

Comment:
Shouldn't we aim for this to be `false` by default?

Reply:
It doesn't really matter whether it's true or false from a functional standpoint. But from a performance standpoint, I think that using `transpose_tensor = true` might be better.

When the tensor is copied into a contiguous space with `tensor_ = tensor.clone(at::MemoryFormat::Contiguous)`, it will be in row-major order. cuBLAS requires column-major order, which is why we need `transpose_tensor = true`.

We could alternatively copy into a column-major order space with `tensor_ = tensor.transpose().clone(at::MemoryFormat::Contiguous).transpose()`, and then set `transpose_tensor = false`. But those two transpose operations will take extra time, so in my opinion we should avoid them. Does that make sense?

Reply:
Didn't know cuBLAS requires column-major. That makes it all clear. Thanks.

Reply:
Yeah, it's not the intuitive choice, but the cuBLAS manual says that using column-major simplifies interfacing with fortran applications.

Reply:
To be clear, cublas does not require column-major, and performance is not universally better with column-major. Cublas will take whatever matrices it is given, but then transa and transb arguments have to be setup correctly, and this is what this function is doing. trans[a,b]='n' means that matrix is column-major, which is opposite of what non-transposed matrix would mean in other languages and pytorch.